src/reanimated2/core.ts

```diff
@@ -213,6 +213,22 @@ if (!isWeb() && isConfigured()) {
       info: runOnJS(capturableConsole.info),
     };
     _setGlobalConsole(console);
+
+    const nativeRequestAnimationFrame = global.requestAnimationFrame;
+    let callbacks: Array<(time: number) => void> = [];
+    global.requestAnimationFrame = (
+      callback: (time: number) => void
+    ): number => {
+      callbacks.push(callback);
+      if (callbacks.length === 1) {
```

Comment:
I know this is obvious but maybe let's add a comment that explains the magic (we schedule a function that captures `callbacks` array which that is filled in subsequent calls (batching) but since the function needs to be executed only once we schedule it only when length equals 1 meaning that before pushing it was empty).